src/Serializers/Orleans.Serialization.Protobuf/ProtobufCodec.cs

```diff
@@ -95,16 +97,34 @@ bool IGeneralizedCopier.IsSupportedType(Type type)
     /// <inheritdoc/>
     bool? ITypeFilter.IsTypeAllowed(Type type)
     {
-        if (!typeof(IMessage).IsAssignableFrom(type))
+        if (!MessageType.IsAssignableFrom(type))
         {
             return null;
         }
 
+        if (type == MessageType)
+        {
+            // While IMessage is the basis of all supported types, it isn't directly supported
+            return null;
+        }
+
         return ((IGeneralizedCodec)this).IsSupportedType(type) || ((IGeneralizedCopier)this).IsSupportedType(type);
-    } 
+    }
 
     private static bool IsProtobufMessage(Type type)
     {
+        if (type == MessageType)
+        {
+            // Not a concrete implementation, so not directly serializable
+            return false;
```

Comment:
Maybe just returning `true` here and not changing `IsTypeAllowed` would make more sense.